crates/evm/src/lib.rs

```diff
@@ -241,6 +241,18 @@ pub trait TransactionEnv:
         self
     }
 
+    /// Returns the configured nonce.
+    fn nonce(&self) -> Option<u64>;
+
+    /// Sets the nonce.
+    fn set_nonce(&mut self, nonce: Option<u64>);
+
+    /// Sets the nonce.
+    fn with_nonce(mut self, nonce: Option<u64>) -> Self {
```

Comment:
all of these make sense.

can we add a note here why this accepts None

should we add `without_nonce(self)`?

Reply:
yeah can do, `without_nonce` or `unset_nonce` might be good